src/OrchardCore/OrchardCore.Users.Abstractions/Events/LoginFormEventBase.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.AspNetCore.Mvc;
+
+namespace OrchardCore.Users.Events;
+
+public abstract class LoginFormEventBase : ILoginFormEvent
```

Comment:
Why do we need another base class, this could be in the interface.

Reply:
because most implementations don't need to override all of the methods defined in the interface. having an abstract class will allow you to implement the methods that you want without having to implement all of them. And in the future if we need to add methods to the interface, it'll reduce the chance of breaking change in more projects.

Reply:
I wrote: "this could be in the interface"

You did exactly that for the new method

Reply:
Yes I did that to avoid breaking change in 2.1. But the default implementation will be removed in 3.0 from the interfaces https://github.com/OrchardCMS/OrchardCore/issues/16905

Reply:
Out of curiosity, what's wrong with DIM here? These are events, so it's unlikely you'll ever provide a "real" base implementation anyway.

Reply:
> what's wrong with DIM here?

This maybe a good question to add on #16905.  But generally, it's weird to have a default implementation on an interface from principal stand point. They are very useful to avoid breaking changes. In major release when we can introduce a breaking change, we should eliminate them.

@sebastienros can we leave the base class here and merge this PR or are you requesting that I remove it?